packages/react-components/src/components/ChatMessageComponent.tsx

```diff
@@ -0,0 +1,312 @@
+  const [isEditing, setIsEditing] = useState(false);
+
+  const menuClass = mergeStyles(chatActionsCSS, { 'ul&': { backgroundColor: theme.palette.white } });
+
+  const actionMenu: MenuProps = useMemo(
+    (): MenuProps => ({
+      iconOnly: true,
+      activeIndex: -1,
+      className: menuClass,
+      items: [
+        {
+          children: (
+            <MoreMenu
+              onEditClick={() => {
+                setIsEditing(true);
+              }}
+              onRemoveClick={async () => {
+                onDeleteMessage && message.payload.messageId && (await onDeleteMessage(message.payload.messageId));
+              }}
+            />
+          ),
+
+          key: 'menuButton2',
+          'aria-label': 'More options',
+          indicator: false
+          // eslint-disable-next-line @typescript-eslint/no-empty-function
+        }
+      ]
+    }),
+    [menuClass, message.payload.messageId, onDeleteMessage]
+  );
+
+  if (message.type !== 'chat') {
+    return <></>;
+  }
+
+  const payload: ChatMessagePayload = message.payload;
+
+  if (isEditing) {
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    return (
+      <EditBox
+        initialValue={payload.content ?? ''}
+        onSubmit={async (text) => {
+          onUpdateMessage && payload.messageId && (await onUpdateMessage(payload.messageId, text));
+          setIsEditing(false);
+        }}
+        onCancel={() => {
+          setIsEditing(false);
+        }}
+      />
+    );
+  }
+  const messageContentItem = GenerateMessageContent(payload);
+  return (
+    <Chat.Message
+      className={mergeStyles(chatMessageStyle as IStyle, messageContainerStyle as IStyle)}
+      content={messageContentItem}
+      author={<Text className={mergeStyles(chatMessageDateStyle as IStyle)}>{payload.senderDisplayName}</Text>}
+      mine={payload.mine}
+      timestamp={
+        <Text data-ui-id={ids.messageTimestamp}>
+          {payload.createdOn
+            ? showDate
+              ? formatTimestampForChatMessage(payload.createdOn, new Date(), strings)
+              : formatTimeForChatMessage(payload.createdOn)
+            : undefined}
+        </Text>
+      }
+      positionActionMenu={false}
+      actionMenu={
+        !editDisabled && message.payload.status !== 'sending' && message.payload.mine ? actionMenu : undefined
+      }
+    />
+  );
+};
+
+const onRenderCancelIcon = (color: string): JSX.Element => {
+  const className = mergeStyles(inputBoxIcon, { color });
+  return <Icon iconName={'EditBoxCancel'} className={className} />;
+};
+
+const onRenderSubmitIcon = (color: string): JSX.Element => {
+  const className = mergeStyles(inputBoxIcon, { color });
+  return <Icon iconName={'EditBoxSubmit'} className={className} />;
+};
+
+type EditBoxProps = {
+  onCancel?: () => void;
+  onSubmit: (text: string) => void;
+  initialValue: string;
+};
+
+const EditBox = (props: EditBoxProps): JSX.Element => {
```

Comment:
Given that you are already refactoring and dividing the chat thread component into smaller pieces. Would make sense to give Edit box component it's individual file too. It will get fairly complex in the future and we will have to eventually do it. ;)